data-prepper-plugins/s3-sink/src/main/java/org/opensearch/dataprepper/plugins/sink/s3/PredefinedObjectMetadata.java

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.sink.s3;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+public class PredefinedObjectMetadata {
+    @JsonProperty("number_of_objects")
+    private String numberOfObjects;
```

Comment:
Why is this a String? Sounds like it would be integer with the name

Reply:
Maybe we should call this `number_of_objects_key`. Will do it in the next PR